### PR TITLE
clean up some version_info checks

### PR DESCRIPTION
**Pull request by matthiasdiener**
(minimum required version is 3.8 currently)

### CODE REVIEW

File: pytools/tag.py [hunk 1]
```diff
@@ -378,23 +377,18 @@ class IgnoredForEqualityTag(Tag):
...
-else:
-    TagsType = FrozenSet[Tag]
-    TagOrIterableType = ToTagSetConvertible
-    T_co = TypeVar("TaggableT", bound="Taggable")
+def __getattr__(name):
+    replacement_and_obj = _depr_name_to_replacement_and_obj.get(name, None)
+    if replacement_and_obj is not None:
+        replacement, obj, year = replacement_and_obj
+        from warnings import warn
+        warn(f"'arraycontext.{name}' is deprecated. "
```
Comment (matthiasdiener) [inline pytools/tag.py:385]: should this be

```suggestion
        warn(f"'pytools.tag.{name}' is deprecated. "
```

? (same below)

File: pytools/test/test_pytools.py [hunk 2]
```diff
@@ -31,7 +31,6 @@
 from typing import FrozenSet
 
 
-@pytest.mark.skipif("sys.version_info < (2, 5)")
```
Comment (alexfikl) [inline pytools/test/test_pytools.py:34]: Did this even work? I thought the first argument had to be a bool.

File: pytools/tag.py (hunk 1 above)
Comment (alexfikl) [reply pytools/tag.py:385]: Yeah, definitely! This was just copy pasted from a similar `arraycontext` change from what I recall.

File: pytools/test/test_pytools.py (hunk 2 above)
Comment (matthiasdiener) [reply pytools/test/test_pytools.py:34]: I *think* it worked, see e.g. here: https://docs.pytest.org/en/7.1.x/historical-notes.html#string-conditions

File: pytools/tag.py (hunk 1 above)
Comment (matthiasdiener) [reply pytools/tag.py]: Changed in [6dd3883](https://github.com/inducer/pytools/pull/218/commits/6dd3883b1edff586e3f8b2bbd7ed06091695d987)

File: pytools/test/test_pytools.py (hunk 2 above)
Comment (alexfikl) [reply pytools/test/test_pytools.py:34]: Oooh! Neat!

**Comment by inducer** [comment]
Thx!

**Pull request merged**